SPECS/mariner-rpm-macros/gen-ld-script.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+echo "gen-ld-script.sh generating linker script"
+
+# 
+# Embed the following metadata:
+#
+# {
+#  "os": "mariner",
+#  "osVersion": "1.0",
+#  "type": "rpm"
+# }
+
+LINKER_SCRIPT_START="SECTIONS { .note.package ALIGN(4): { "
+LINKER_SCRIPT_NAMESZ="BYTE(0x04) BYTE(0x00) BYTE(0x00) BYTE(0x00) "
```

Comment:
Can we get additional comments around each of these....so we can see what the descriptions actually say/embed?

Reply:
Sure, I've added comments